src/main/java/ml/shifu/shifu/enhancer/EnhancerFactory.java

```diff
@@ -0,0 +1,96 @@
+package ml.shifu.shifu.enhancer;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * EnhancerFactory is used to generate Enhancer according to the name.
+ * <p>
+ * Author: Wu Devin (haifwu@paypal.com)
+ * Date: 20/01/2018
+ */
+public class EnhancerFactory {
+    private static Map<EnhanceType, EnhanceCallable> enhancerTypeToEnhancerMap = new HashMap<EnhanceType, EnhanceCallable>(10);
+    private static EnhancerFactory instance = new EnhancerFactory();
+
+    private EnhancerFactory(){
+    }
+
+    public enum EnhanceType {
+        LOG("log"),
+        SQRT("sqrt"),
+        SQUARE("square");
+
```

Comment:
can we also support sqrt(x+1), log(x+1) like feature which may be called to LOGPLUS1, SQRTPLUS1?